src/goto-instrument/full_slicer.cpp

```diff
@@ -20,9 +20,11 @@ Author: Daniel Kroening, kroening@kroening.com
 void full_slicert::add_dependencies(
   const cfgt::nodet &node,
   queuet &queue,
-  const dependence_grapht &dep_graph,
+  dependence_grapht &dep_graph,
   const dep_node_to_cfgt &dep_node_to_cfg)
 {
+  // node.PC may not have been previously explored
+  // so dep_graph cannot be const as this may create new domains
```

Comment:
Yes at the moment.  If your patch for storing only merge points gets the obvious extension to do on-the-fly computation of missing domains then it will start to be a serious amount of computation and I'd feel bad about hiding that behind a mutable.

Reply:
new states